int64 imgX
int64 imgY